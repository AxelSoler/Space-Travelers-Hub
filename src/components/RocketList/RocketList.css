.all-rockets {
  padding-top: 1rem;
}