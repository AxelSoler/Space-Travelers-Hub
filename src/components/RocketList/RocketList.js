import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRockets } from '../../redux/rockets/rockets';
import Rockets from '../Rockets/Rockets';

const RocketList = () => {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  return (
    <ul>
      {
        rockets.map((rocket) => (
          <Rockets
            key={rocket.id}
            name={rocket.name}
            img={rocket.img}
            description={rocket.description}
          />
        ))
      }
    </ul>
  );
};

export default RocketList;
