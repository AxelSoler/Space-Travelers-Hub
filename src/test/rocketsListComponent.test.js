import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RocketList from '../components/RocketList/RocketList'
describe('Rocket List', () => {
  it('Renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <RocketList />
          </Provider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})