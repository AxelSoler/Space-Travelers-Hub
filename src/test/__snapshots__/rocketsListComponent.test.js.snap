// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket List Renders correctly 1`] = `
<ul
  className="all-rockets"
/>
`;
